### PR TITLE
implement exploration page ui

### LINKED ISSUE #294: Exploration page UI
As part of #289 a user interface using the exploration store implemented in #291 needs to be created

### PR DESCRIPTION
**Pull request by derwehr**
closes #294

### CODE REVIEW

File: frontend/src/exploration/questions/Questions.tsx [hunk 1]
```diff
@@ -1,20 +1,122 @@
...
+const selections: Record<string, ExplorationSelection> = {};
+explorationQuestions.forEach((q, qIndex) => {
+  q.answers.forEach((a, aIndex) => {
+    selections[`explorationAnswer-${qIndex}-${aIndex}`] = {
+      answer: a,
+      checked: false,
+    };
+  });
+});
+
```
Comment (joluj) [inline frontend/src/exploration/questions/Questions.tsx:51]: Tracking of checked answers can be done by checking if they are in the store state.
```suggestion
```
See next suggestions

File: frontend/src/exploration/questions/Questions.tsx [hunk 2]
```diff
@@ -1,20 +1,122 @@
...
 function Questions(): JSX.Element {
   const classes = useStyle();
-  return <Box className={`${classes.container} Questions`}>TODO</Box>;
+  const explorationStore = useService(ExplorationStore);
+  const [state, setState] = useState(selections);
```
Comment (joluj) [inline frontend/src/exploration/questions/Questions.tsx:55]:
```suggestion
  const answers = useObservable(
    explorationStore.getState(),
    explorationStore.getValue()
  );
```

File: frontend/src/exploration/questions/Questions.tsx [hunk 3]
```diff
@@ -1,20 +1,122 @@
...
+   * @param event the changeEvent emitting this function.
+   */
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, checked } = event.target;
+    const { answer } = state[name];
+
+    setState({
+      ...state,
+      [name]: { answer, checked },
+    });
```
Comment (joluj) [inline frontend/src/exploration/questions/Questions.tsx:70]:
```suggestion
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>, answer: ExplorationAnswer) => {
```

File: frontend/src/exploration/questions/Questions.tsx [hunk 4]
```diff
@@ -1,20 +1,122 @@
...
+              <FormControl className={classes.formControl}>
+                <FormGroup>
+                  {question.answers.map((answer, aIndex) => (
+                    <FormControlLabel
+                      control={
+                        <Checkbox
+                          checked={
+                            state[`explorationAnswer-${qIndex}-${aIndex}`]
+                              .checked
+                          }
```
Comment (joluj) [inline frontend/src/exploration/questions/Questions.tsx:103]:
```suggestion
                          checked={answers.some((a) => a === answer)}
```
Simply checks if answer is in selected answers from store

File: frontend/src/exploration/questions/Questions.tsx [hunk 5]
```diff
@@ -1,20 +1,122 @@
...
+                          onChange={handleChange}
+                          name={`explorationAnswer-${qIndex}-${aIndex}`}
```
Comment (joluj) [inline frontend/src/exploration/questions/Questions.tsx:105]:
```suggestion
```

It's now unnecessary i think

**Review by joluj — CHANGES_REQUESTED** [review]
Looks good and works.
However i think the handling is a bit more complicated than necessary. I have provided suggestions that work (cleanup still missing).
A single line of tsdoc with would also be nice for the component, can also be just a link to the user stoy.

File: frontend/src/exploration/questions/Questions.tsx (hunk 5 above)
Comment (derwehr) [reply frontend/src/exploration/questions/Questions.tsx:105]: This, together with saving the answers in the state was needed to remove answers on un-ticking their checkboxes in handleChange (line 64).
Is there a way to do that without saving the answers, that I'm not seeing right now?

Comment (joluj) [reply frontend/src/exploration/questions/Questions.tsx:109]: Ah i missed the change suggestion for onOhange:
```ts
onChange{ (event) => handleChange(event, answer) } // answer comes from the loop. store handles answers by reference.
```

Comment (derwehr) [reply frontend/src/exploration/questions/Questions.tsx]: Oh, i see thank you

**Review by joluj — APPROVED** [review]
LGTM.
Compiler still complains about an unused variable

**Pull request merged**